pyartcd/pyartcd/pipelines/cleanup_locks.py

```diff
@@ -1,22 +1,38 @@
-import click
+from aioredlock import Lock
 
-from pyartcd import redis
+from pyartcd import redis, jenkins, constants
 from pyartcd.cli import cli, click_coroutine, pass_runtime
+from pyartcd.locks import LockManager
 from pyartcd.runtime import Runtime
 
 
 @cli.command('cleanup-locks')
-@click.option('--locks', required=True, help='Locks to be released on Redis')
 @pass_runtime
 @click_coroutine
-async def cleanup_locks(runtime: Runtime, locks: str):
-    locks_to_release = filter(lambda arg: arg, locks.split(','))
+async def cleanup_locks(runtime: Runtime):
+    lock_manager = LockManager([redis.redis_url()])
+    active_locks = await lock_manager.get_locks()
 
-    for lock_name in locks_to_release:
-        runtime.logger.warning('Deleting lock %s', lock_name)
-        res = await redis.delete_key(lock_name)
+    try:
+        for lock_name in active_locks:
+            # Lock ID is a build URL minus Jenkins server base URL
+            build_path = await lock_manager.get_lock_id(lock_name)
+            build_url = f'{constants.JENKINS_UI_URL}/{build_path}'
+            runtime.logger.info('Found build %s associated with lock %s', build_url, lock_name)
```

Comment:
I think this can be wrong, when build_path is not a valid build path?

Reply:
That will cause jenkinsapi to raise an exception, that's caught [here](https://github.com/openshift-eng/art-tools/pull/38/files#diff-30b25a19eea9451c6fffe6c527a6343545fb9e06c5949d1a5c4c8206a5f2bf89R35). The job will log the build that couldn't be fetched, and continue.

This shouldn't happen once old locks are gone, as new ones will always point to a valid Jenkins build

Reply:
Then I would move the log message to after we know it's a valid build
The message, like in hackjob, "Found build https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/7422cbdd-37fc-4d1e-8deb-c701ee6b8dcb associated with lock build-lock-4.14" would be confusing

Reply:
Hm I think we should do some input validation on the value before constructing and calling it as part of a url, as a good security practice too - I don't know if some url code injection attack exists for jenkins - but yeah just to be safe and follow input validation practices